browser/resources/settings/brave_wallet_page/brave_wallet_page.ts

```diff
@@ -180,6 +223,19 @@ class SettingsBraveWalletPage extends SettingsBraveWalletPageBase {
     this.browserProxy_.resetTransactionInfo()
     window.alert(this.i18n('walletResetTransactionInfoConfirmed'))
   }
+
+  onClearPinnedNftTapped_() {
+    if (this.pinnedNftCount_ == 0) {
+      return
+    }
+    var message = this.i18n('walletClearPinnedNftConfirmation')
+    if (window.prompt(message) !== this.i18n('walletResetConfirmationPhrase'))
+      return
+    this.browserProxy_.clearPinnedNft().then(val => {
+      this.onShowOptionChanged_()
+    })
+    window.alert(this.i18n('walletClearPinnedNftConfirmed'))
```

Comment:
Should the alert show when the clearPinnedNFT is completed instead of immediately after the prompt is closed?

Reply:
I am not sure about this - i don't want to subscribe on cleanup operation result since it may freeze and block the UI, on other hand this operation is done quickly and user can see the result in the setting's counter. So if it fails pinned NFT count will stay unchanged.

Reply:
What is the point of showing a confirmation message if it happens immediately after the previous message? Or what is the point of having `onShowOptionChanged_()` async after `clearPinnedNft` completes, but not the alert?
Are you worried about the promise not getting resolved if there's an error? If so, you can cause the promise to be rejected, or otherwise make sure any error is handled correctly in the C++. I would be more worried the browser will crash and not that the alert won't show.

Reply:
I have removed  window.alert(this.i18n('walletClearPinnedNftConfirmed'))